src/snowflake/snowpark/modin/plugin/_internal/align_utils.py

```diff
@@ -93,3 +97,120 @@ def align_axis_0_right(
             right_frame_data_ids + right_index_ids
         )
     return right_result, right_frame, right_frame_data_ids, right_index_ids
+
+
+def align_axis_1(
+    frame1: InternalFrame,
+    frame2: InternalFrame,
+    join: Literal["inner", "outer", "left", "right"],
+    sort: Optional[bool] = False,
+) -> tuple[InternalFrame, InternalFrame]:
+    """
+    Concatenate frames on index axis by taking using UNION operator.
+    Snowflake identifiers of output frame are based on snowflake identifiers from first
+    frame.
+    Args:
+        frame1: First frame
+        frame2: Second frame
+        join: How to handle column index
+            'inner': Output frame contains only overlapping columns from both frames.
+            'outer': Output frame contains union of columns from both frames.
+            'left': Output frame contains columns from left frame.
+            'right': Output frame contains columns from right frame.
+        sort: Sort column axis if True.
+
+    Returns:
+        New InternalFrame after taking union of given frames.
+    """
+    columns1 = frame1.data_columns_index
+    columns2 = frame2.data_columns_index
+
+    if join == "inner":
+        # Preserves the order from calling index.
+        # For example:
+        # pd.Index([3, 1, 2]).intersection(pd.Index([1, 2, 3]) will result in
+        # pd.Index([3, 1, 2])
+        data_column_labels = columns1.intersection(columns2, sort=False)
+    elif join == "outer":
+        # Preserves the order from calling index. And for labels not in calling index
+        # preserves the order from argument index.
+        # For example:
+        # pd.Index([3, 1, 2]).union(pd.Index([1, 4, 2, 3, 5]) will result in
+        # pd.Index([3, 1, 2, 4, 5])
+        data_column_labels = columns1.union(columns2, sort=False)
+    elif join == "left":
+        data_column_labels = columns1
+    elif join == "right":
+        data_column_labels = columns2
+    else:
+        raise AssertionError(
+            f"Invalid join type '{join}'. Accepted values are 'inner', 'outer', 'left', and 'right'."
+        )
+    if sort:
+        data_column_labels = data_column_labels.sort_values()
+
+    frame1 = _select_columns(frame1, data_column_labels.tolist())
+    frame2 = _select_columns(frame2, data_column_labels.tolist())
+    return frame1, frame2
+
+
+def _select_columns(
```

Comment:
should we add a unit test for _select_columns ?

Reply:
usually I believe internal util functions don't have unit tests, such as concat_utils._select_columns but I can add one!